python/pyspark/sql/readwriter.py

```diff
@@ -234,6 +234,8 @@ def json(self, path, schema=None, primitivesAsString=None, prefersDecimal=None,
                                 default value, ``yyyy-MM-dd'T'HH:mm:ss.SSSXXX``.
         :param multiLine: parse one record, which may span multiple lines, per file. If None is
                           set, it uses the default value, ``false``.
+        :param lineSep: defines the line separator that should be used for parsing. If None is
+                        set, it uses the default value, ``\\r\\n`` or ``\\n``.
```

Comment:
It has a different value for reading and writing individually. I tried to not change both. So, the related pointers for JSON read should be ...

https://github.com/apache/spark/pull/18581/files#diff-5ac20b8d75a20117deaa9ba4af814090R142 for data

https://github.com/apache/spark/pull/18581/files#diff-5ac20b8d75a20117deaa9ba4af814090R132 for schema inference.